tests/docs_test.go

```diff
@@ -0,0 +1,175 @@
+package tests
+
+import (
+	"context"
+	"testing"
+
+	orbitdb2 "berty.tech/go-orbit-db"
+	. "github.com/smartystreets/goconvey/convey"
+)
+
+func TestDocumentsStore(t *testing.T) {
+	tmpDir, clean := testingTempDir(t, "db-docsstore")
+	defer clean()
+
+	cases := []struct{ Name, Directory string }{
+		{Name: "in memory", Directory: ":memory:"},
+		{Name: "persistent", Directory: tmpDir},
+	}
+
+	for _, c := range cases {
+		t.Run(c.Name, func(t *testing.T) {
+			testingDocsStore(t, c.Directory)
+		})
+	}
+}
+
+func createDocument(id string, key string, value string) map[string]interface{} {
+	m := make(map[string]interface{})
+	m["_id"] = id
+	m[key] = value
+	return m
+}
+
+func testingDocsStore(t *testing.T, dir string) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	dbname := "orbit-db-tests"
+	Convey("orbit-db - Documents Database", t, FailureHalts, func(c C) {
+		mocknet := testingMockNet(ctx)
+
+		node, clean := testingIPFSNode(ctx, t, mocknet)
+		defer clean()
+
+		db1IPFS := testingCoreAPI(t, node)
+
+		orbitdb1, err := orbitdb2.NewOrbitDB(ctx, db1IPFS, &orbitdb2.NewOrbitDBOptions{
+			Directory: &dir,
+		})
+		defer orbitdb1.Close()
+
+		c.So(err, ShouldBeNil)
+
+		db, err := orbitdb1.Docs(ctx, dbname, nil)
+		c.So(err, ShouldBeNil)
+
+		defer db.Close()
+
+		c.Convey("creates and opens a database", FailureHalts, func(c C) {
+			db, err := orbitdb1.Docs(ctx, "first docs database", nil)
+			c.So(err, ShouldBeNil)
+
+			if db == nil {
+				t.Fatalf("db should not be nil")
+			}
+
+			defer db.Close()
+
+			c.So(db.Type(), ShouldEqual, "docstore")
+			c.So(db.DBName(), ShouldEqual, "first docs database")
+		})
+
+		c.Convey("put", FailureHalts, func(c C) {
+			document := createDocument("doc1", "hello", "world")
+			_, err := db.Put(ctx, document)
+			c.So(err, ShouldBeNil)
+
+			docs, err := db.Get(ctx, "doc1", true)
+			c.So(err, ShouldBeNil)
+			c.So(len(docs), ShouldEqual, 1)
+			c.So(docs[0], ShouldResemble, document)
+		})
+
+		c.Convey("get", FailureHalts, func(c C) {
+			document := createDocument("doc1", "hello", "world")
+			_, err := db.Put(ctx, document)
+			c.So(err, ShouldBeNil)
+
+			docs, err := db.Get(ctx, "doc1", true)
```

Comment:
Can you ensure that retrieving entries without case-sensitivity works? Specifically I had issues with a Get() partial match